rllib/env/single_agent_episode.py

```diff
@@ -128,7 +126,7 @@ def concat_episode(self, episode_chunk: "SingleAgentEpisode"):
             from both episodes.
         """
         assert episode_chunk.id_ == self.id_
-        assert not self.is_done
+        assert not self.is_done and not self.is_numpy
```

Comment:
For simplicity, we assume that Episode is still in the "list-format" (not numpyized yet).
We might have to change this API in the future, but right now, it's only used inside DreamerV3's replay buffer anyways (and in some test cases).